index.bs

```diff
@@ -55,13 +55,326 @@ before a user has picked a location to save to, without forcing the website to u
 different storage mechanism with a different API for such files. The entry point for this is the
 {{StorageManager/getDirectory()|navigator.storage.getDirectory()}} method.
 
+It also defines an API to listen to file system change events. Without it, sites
+can recursively poll the file system to find changes in the files or folder
+structure. This can be time consuming especially for large directories. The
+{{FileSystemObserver}} provides an API to listen for change events so that
+polling for them isn't required.
+
 
 # Files and Directories # {#files-and-directories}
 
 ## Concepts ## {#concepts}
 
+### File System ### {#concept-file-system}
+
+A <dfn export>file system</dfn> is an [=implementation-defined=]
+[=storage endpoint=] that maintains a mapping of [=file system path=]s to
```

Comment:
I don't think it makes sense for a "file system" to be a storage endpoint. A storage endpoint is defined as "an API" after all. The bucket file system somehow should still be a storage endpoint, but a generic file system concept here I don't think should have anything to do with a "storage endpoint".